File: nom-ui/src/app/core/models/macro-goal.model.ts
export interface MacroGoal {
  caloriesTarget: number | null;
  proteinGramsTarget: number | null;
  carbGramsTarget: number | null;
  fatGramsTarget: number | null;
}

export interface EffectiveMacroGoal extends MacroGoal {
  /** Where the effective targets came from. */
  source: 'person' | 'household' | 'none';
}

File: nom-api/Nom.Orch/Models/Plan/MacroGoalModel.cs
namespace Nom.Orch.Models.Plan
{
    /// <summary>
    /// Daily macronutrient targets for a person or a household. Null means no
    /// target is set for that macro.
    /// </summary>
    public class MacroGoalModel
    {
        public decimal? CaloriesTarget { get; set; }
        public decimal? ProteinGramsTarget { get; set; }
        public decimal? CarbGramsTarget { get; set; }
        public decimal? FatGramsTarget { get; set; }

        public bool HasAnyTarget =>
            CaloriesTarget.HasValue || ProteinGramsTarget.HasValue
            || CarbGramsTarget.HasValue || FatGramsTarget.HasValue;
    }
}
